### PR TITLE
Fix #92: Drop support for useless `optimize` option

### LINKED ISSUE #92: Drop the "size" optimize mode
As I understand, David studied the construction of compilers, that is why he create a library with two possible implementations:
- statically generated functions ("speed" optimizing mode)
- interpreted bytecode ("size" optimizing mode)

Frankly speaking, size of the generated parsers in the "size" mode don't differ much from these parsers, generated in the "speed" mode, but their parsing speed is worse.

Generating in the "size" mode would be valuable if the generator had generated only bytecode instructions; but it is also generating the interpreter, which for small grammars don't differ much in the size from the grammars, generated in the "speed" mode.

Judging that, there is no reasons to keep the "size" mode, so I suggest to drop it. I think, that nobody uses that mode.

### PR DESCRIPTION
**Pull request by Mingun**
There no reasons for use value other than `speed`.

CLI option and definition in `.d.ts` are left for backward compatibility

### CODE REVIEW

File: bin/peggy [hunk 1]
```diff
@@ -33,8 +33,6 @@ function printHelp() {
   console.log("      --format <format>              format of the generated parser: amd,");
   console.log("                                     commonjs, globals, umd (default: commonjs)");
   console.log("  -h, --help                         print help and exit");
-  console.log("  -O, --optimize <goal>              select optimization for speed or size");
```
Comment (hildjj) [inline bin/peggy:36]: I have a rewrite of the CLI that I'm still tweaking, so let's double-check that I get the same results there that you have here: `--optimize` is hidden from the help text, and causes a warning on stderr if used.

**Review by hildjj — COMMENTED** [review]
I was assuming we would go ahead and rip the functionality out of generate-js also?

File: README.md [hunk 2]
```diff
@@ -274,7 +270,7 @@ a _per-parse initializer_:
 {
   if (options.multiplier) {
     input = "(" + input + ")*(" + options.multiplier + ")";
-  }
+}
```
[the remark this answers is not in the document]
Comment (Mingun) [reply README.md:273]: Yes. rebase costs + enable "ignored whitespace" setting in the diff viewer

File: bin/peggy (hunk 1 above)
Comment (Mingun) [reply bin/peggy:36]: Yes, all correct

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mingun) [reply README.md]: Fixed

**Comment by hildjj** [comment]
This looks good.  One last thing to discuss.  Should we issue a DeprecationWarning for people that use "size" through the API?  It would be nice, but I don't want to have to mock `process.emitWarning` in the web build.

**Comment by Mingun** [comment]
Not enough `@deprecated` comment in the `.d.ts`?  I think, that modern users would, at least, use typescript types and should see that warning.

**Comment by hildjj** [comment]
I'm more worried about the non-modern users here, I think, but I see your point.  Let's land this then be on the lookout for issues that get filed if people are confused.  I think we should land this now since I want to rebase other changes on top of it.  Make sense?

**Comment by Mingun** [comment]
Make sense.

**Comment by StoneCypher** [comment]
> Not enough @deprecated comment in the .d.ts? I think, that modern users would, at least, use typescript types and should see that warning.

This isn't how typescript works.

**Pull request merged**